Automattic/vip-go-mu-plugins
# Replace CircleCI with GitHub Actions

We are moving away from CircleCI to GitHub Actions.

Linting (1):
  - Lint PHP files (**php-lint**)

Core Tests: PHP 7.4, latest WordPress (2):
  - Run WP Core Tests (WordPress latest, multisite: no) (**php74-core-tests**)
  - Run WP Core Tests (WordPress latest, multisite: yes) (**php74-core-multisite-tests**)

JetPack, Single-Site, PHP 7.4 (7):
  - WP 5.5.x, MS: no, JP: yes, PHP: 7.4 (**php74-build-singlesite-55**)
  - WP 5.6.x, MS: no, JP: yes, PHP: 7.4 (**php74-build-singlesite-56**)
  - WP 5.7.x, MS: no, JP: yes, PHP: 7.4 (**php74-build-singlesite-57**)
  - WP 5.8.x, MS: no, JP: yes, PHP: 7.4 (**php74-build-singlesite-58**)
  - WP 5.9.x, MS: no, JP: yes, PHP: 7.4 (**php74-build-singlesite-59**)
  - WP latest, MS: no, JP: yes, PHP: 7.4 (**php74-build-singlesite**)
  - WP nightly, MS: no, JP: yes, PHP: 7.4

JetPack, Multi-Site, PHP 7.4 (2):
  - WP latest, MS: yes, JP: yes, PHP: 7.4 (**php74-build-multisite**)
  - WP nightly, MS: yes, JP: yes, PHP: 7.4

No JetPack, WP latest, PHP 7.4 (2):
  - WP latest, MS: no, JP: no, PHP: 7.4 (**php74-build-singlesite-no-jetpack**)
  - WP latest, MS: yes, JP: no, PHP: 7.4 (**php74-build-multisite-no-jetpack**)

PHP 8.0, JetPack (4):
  - WP latest, MS: no, JP: yes, PHP: 8.0
  - WP latest, MS: yes, JP: yes, PHP: 8.0
  - WP nightly, MS: no, JP: yes, PHP: 8.0
  - WP nightly, MS: yes, JP: yes, PHP: 8.0

Parsely (9):
  - 2.6 (see #3165):
    - Parse.ly: 2.6 filter_enabled
    - Parse.ly: 2.6 option_enabled
  - 3.1:
    - Parse.ly: 3.1 filter_enabled
    - Parse.ly: 3.1 option_enabled
  - 3.2 (see #3140):
    - Parse.ly: 3.2 filter_enabled
    - Parse.ly: 3.2 option_enabled
  - 3.3:
    - Parse.ly: 3.3 filter_enabled
    - Parse.ly: 3.3 option_enabled
    - Parse.ly: 3.3 filter_and_option_enabled

Search Dev Tools (2):
  - Lint Search Dev Tools
  - Build Search Dev Tools (**search-dev-tools**)

Misc (1):
  - E2E tests
  - Deploy (**push-code-to-build**)

Jobs to migrate:
  - ~~create-changelog-draft~~
